src/App.js

```diff
@@ -87,252 +96,270 @@ class Root extends React.Component {
       LOCAL_STORE.setNetwork('mainnet');
     }
 
-    if (IPC_RENDERER) {
-      // Event called when user quits hathor app
-      IPC_RENDERER.on('ledger:closed', async () => {
-        const storage = LOCAL_STORE.getStorage();
-        if (
-          storage &&
-          await LOCAL_STORE.isLoaded() &&
-          await storage.isHardwareWallet()
-        ) {
-          this.props.updateLedgerClosed(true);
-        }
-      });
-
-      IPC_RENDERER.on('ledger:manyTokenSignatureValid', async (_, arg) => {
-        const storage = LOCAL_STORE.getStorage();
-        if (
-          storage &&
-          await storage.isHardwareWallet()
-        ) {
-          // remove all invalid signatures
-          // arg.data is a list of uids with invalid signatures
-          arg.data.forEach(uid => {
-            tokens.removeTokenSignature(uid.toString('hex'));
-          });
-        }
-      });
+    // If there is no ledger connected, finish execution here
+    if (!IPC_RENDERER) {
+      return;
     }
-  }
 
-  componentWillUnmount() {
-    if (IPC_RENDERER) {
-      IPC_RENDERER.removeAllListeners('ledger:closed');
-      IPC_RENDERER.removeAllListeners('ledger:manyTokenSignatureValid');
+    // Registers the event handlers for the ledger
+    // Event called when user quits hathor app
+    IPC_RENDERER.on('ledger:closed', async () => {
+      const storage = LOCAL_STORE.getStorage();
+      if (
+        storage &&
+        await LOCAL_STORE.isLoaded() &&
+        await storage.isHardwareWallet()
+      ) {
+        dispatch(updateLedgerClosed(true));
+      }
+    });
+
+    IPC_RENDERER.on('ledger:manyTokenSignatureValid', async (_, arg) => {
+      const storage = LOCAL_STORE.getStorage();
+      if (
+        storage &&
+        await storage.isHardwareWallet()
+      ) {
+        // remove all invalid signatures
+        // arg.data is a list of uids with invalid signatures
+        arg.data.forEach(uid => {
+          tokensUtils.removeTokenSignature(uid.toString('hex'));
+        });
+      }
+    });
+
+    // Removes the ledger event listeners on unmount, if necessary
+    return () => {
+      if (IPC_RENDERER) {
+        IPC_RENDERER.removeAllListeners('ledger:closed');
+        IPC_RENDERER.removeAllListeners('ledger:manyTokenSignatureValid');
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    // Fetch the version information before rendering any screens, if possible
+    if (isVersionAllowed === undefined && wallet) {
+      helpersUtils.loadStorageState();
```

Comment:
This is already as close to the initial startup as I could move it while being in context, because it needs an initialized wallet to check the API version.

If I moved this into the `ErrorWrapper`, `ErrorBoundary` it would seem to be out of place, I think.
Do you agree?